airflow-core/src/airflow/api_fastapi/core_api/services/public/task_instances.py

```diff
@@ -239,28 +254,46 @@ def handle_bulk_update(
                     map_index=task_instance_body.map_index,
                     update_mask=None,
                 )
-
                 for key, _ in data.items():
                     if key == "new_state":
-                        _patch_task_instance_state(
+                        updated_tis = _patch_task_instance_state(
                             task_id=task_instance_body.task_id,
                             dag_run_id=self.dag_run_id,
                             dag=dag,
                             task_instance_body=task_instance_body,
                             session=self.session,
                             data=data,
+                            commit=self.commit,
                         )
+                        if updated_tis:
+                            all_updated_tis.extend(updated_tis)
+
                     elif key == "note":
                         _patch_task_instance_note(
                             task_instance_body=task_instance_body, tis=tis, user=self.user
                         )
+                        all_updated_tis.extend(tis)
```

Comment:
That's wrong. Structure is not symetrical between patch `new_state` and path `note`.

The part of `updated_tis` is not correct in this code path. 
```suggestion
                        uptated_tis = all_updated_tis.extend(tis)
```

Reply:
Also missing the `if` statement present above.